libgalois/include/katana/Timer.h

```diff
@@ -104,6 +104,24 @@ class CondStatTimer<false> {
   uint64_t get_usec() const { return 0; }
 };
 
+template <typename T>
+class TimerGuard {
+public:
+  T timer;
+
+  template <typename... Args>
+  TimerGuard(Args&&... args) : timer(std::forward<Args>(args)...) {
+    timer.start();
+  }
+
+  TimerGuard(const TimerGuard&) = delete;
+  TimerGuard(TimerGuard&&) = delete;
+  TimerGuard& operator=(const TimerGuard&) = delete;
+  TimerGuard& operator=(TimerGuard&&) = delete;
+
+  ~TimerGuard() { timer.stop(); }
+};
+
```

Comment:
comment on `timeThis` below: if we add `TimerGuard`, I think it would be good to remove `timeThis` so we have just one way to do this.